Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/censors/dcc/DccQrCodeCensor.kt

```diff
@@ -0,0 +1,270 @@
+        data: DccV1.TestCertificateData,
+        message: CensorContainer
+    ): CensorContainer {
+        var newMessage = message
+
+        data.testCenter?.let {
+            newMessage = newMessage.censor(
+                data.testCenter,
+                "test/testCenter"
+            )
+        }
+
+        newMessage = newMessage.censor(
+            data.testResult,
+            "test/testResult"
+        )
+
+        newMessage = newMessage.censor(
+            data.testType,
+            "test/testType"
+        )
+
+        data.testName?.let {
+            newMessage = newMessage.censor(
+                data.testName,
+                "test/testName"
+            )
+        }
+
+        data.testNameAndManufacturer?.let {
+            newMessage = newMessage.censor(
+                data.testNameAndManufacturer,
+                "test/testNameAndManufacturer"
+            )
+        }
+
+        newMessage = newMessage.censor(
+            data.sampleCollectedAtFormatted,
+            "test/sampleCollectedAtFormatted"
+        )
+
+        newMessage = newMessage.censor(
+            data.sampleCollectedAt.toShortDayFormat(),
+            "test/sampleCollectedAt"
+        )
+
+        newMessage = newMessage.censor(
+            data.targetId,
+            "test/targetId"
+        )
+
+        newMessage = newMessage.censor(
+            data.certificateIssuer,
+            "test/certificateIssuer"
+        )
+
+        newMessage = newMessage.censor(
+            data.uniqueCertificateIdentifier,
+            "test/uniqueCertificateIdentifier"
+        )
+
+        newMessage = newMessage.censor(
+            data.certificateCountry,
+            "test/certificateCountry"
+        )
+
+        return newMessage
+    }
+
+    private fun censorNameData(nameData: DccV1.NameData, message: CensorContainer): CensorContainer {
+        var newMessage = message
+
+        nameData.familyName?.let { fName ->
+            newMessage = newMessage.censor(
+                fName,
+                "nameData/familyName"
+            )
+        }
+
+        newMessage = newMessage.censor(
+            nameData.familyNameStandardized,
+            "nameData/familyNameStandardized"
+        )
+
+        nameData.givenName?.let { gName ->
+            newMessage = newMessage.censor(
+                gName,
+                "nameData/givenName"
+            )
+        }
+
+        nameData.givenNameStandardized?.let { gName ->
+            newMessage = newMessage.censor(
+                gName,
+                "nameData/givenNameStandardized"
+            )
+        }
+
+        return newMessage
+    }
+
+    companion object {
+        private val qrCodeStringsToCensor = ArrayList<String>()
+
+        fun addQRCodeStringToCensor(rawString: String) = synchronized(qrCodeStringsToCensor) {
+            qrCodeStringsToCensor.apply {
+                if (contains(rawString)) return@apply
+                add(rawString)
```

Comment:
Not a blocker, but we could use `mutableSetOf` here, and just call `add` as the set will not allow the same object twice.